sycl/include/CL/sycl/types.hpp

```diff
@@ -317,13 +317,32 @@ template <typename T, int N> struct VectorAlignment {
 
 } // namespace detail
 
+#if defined( __GNUC__ )
+#define ALIGNAS(N) alignas(N)
```

Comment:
I think we should define this macro for any C++11 compiler.

Reply:
MSVC Compiler it does not work for all SYCL cases. 
For vec<double, 16> required alignment is 128 by SYCL spec. But for MSVC Compiler maximum alignment is 64.

Reply:
Okay, so switch the order of the conditions.

Reply:
Updated.

Reply:
I still see the check for `__GNUC__`. AFAIK,  `alignas` is supported by clang. Will I see a warning?
NOTE: we use clang for device compilation.

Reply:
No with clang you don't see a warning.
As I understand clang supports GNU C extensions and define `__GNUC__`.
Yes, `alignas `is supported by clang.

Reply:
`alignas` is not GNU C extension. It's standard C++ feature.
I think we should remove `#if defined( __GNUC__ )`.

Reply:
I never said that

> alignas is not GNU C extension. It's standard C++ feature.

The available alignment depends on the compiler, so we cannot use alignas (128) with any compiler.

Now we have MSVC compiler and gnu friendly compilers(GCC and Clang). For them we have handled this problem. But what about the others? alignas (128) may not work with them.

Reply:
@bader Updated. Please take a look.